### PR TITLE
Fix add new limited company

### PR DESCRIPTION
**Pull request by ztolley**
add company

- [x] Refactor add company flow to follow patterns
- [x] Create updated limited company search call to use new companies house search API
- [x] Create new transformer for companies house results
- [x] Create simplified middleware to coordinate getting results, transforming them and sending them for render

### CODE REVIEW

File: test/unit/apps/companies/controllers/add.test.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ describe('Company add controller', function () {
         },
       }
 
-      companyAddController.getAddStepOne(req, res, next)
+      companyAddController.renderAddStepOne(req, res, next)
```
Comment (yeahfro) [inline test/unit/apps/companies/controllers/add.test.js]: could we get some `context` around these tests?

File: src/apps/search/services.js [hunk 2]
```diff
@@ -130,21 +130,25 @@ function searchForeignCompanies (options) {
   return searchCompanies(optionsUkBasedFalse)
 }
 
-function searchLimitedCompanies (options) {
-  const optionsUkBasedTrue = assign({}, options, { isUkBased: true })
+function searchLimitedCompanies ({ token, searchTerm, page = 1, limit = 10 }) {
+  const queryParams = {
+    offset: (page * limit) - limit,
+    limit,
```
Comment (teneightfive) [inline src/apps/search/services.js]: Shorthand first has been our pattern to date. Easier to scan and read.

File: test/unit/apps/companies/controllers/add.test.js (hunk 1 above)
Comment (ztolley) [reply test/unit/apps/companies/controllers/add.test.js]: The test style is very out of date. I'll revisit the whole test suite for company as it's own pr.

File: src/apps/companies/transformers.js [hunk 3]
```diff
@@ -78,33 +85,75 @@ function transformCompanyToListItem ({
...
+  company_category,
+  company_status,
+  sic_code_1,
+  sic_code_2,
+  sic_code_3,
+  sic_code_4,
+  incorporation_date,
+}) {
+  const address = [
+    capitalize(registered_address_1),
```
Comment (teneightfive) [inline src/apps/companies/transformers.js]: I think we have a package to handle this. Can also use it below for postcode to always uppercase postcode?

Comment (ztolley) [reply src/apps/companies/transformers.js]: Postcode is already uppercase. All the CH data is uppercase, always SHOUTS at you

File: src/apps/search/services.js (hunk 2 above)
Comment (ztolley) [reply src/apps/search/services.js]: I know, I do, I just missed this one

**Pull request merged**